.eleventy.js

```diff
@@ -134,6 +182,22 @@ module.exports = function (config) {
     pathPrefix = process.env.BASEURL
   }
 
+  config.addUrlTransform((page) => {
+    if (page.url.startsWith("/pages/knowledge-base/articles")) {
+      return page.url.replace("/pages/knowledge-base/articles", "/knowledge-base");
+    }
+    if (page.url.startsWith("/pages/pages/knowledge-base/articles")) {
+      return page.url.replace("/pages/pages/knowledge-base/articles", "/pages/knowledge-base");
+    }
+    if (page.url.startsWith("/pages/news/articles/")) {
+      let dateMatches = page.url.match(/[0-9]{4}-[0-9]{2}-[0-9]{2}/g);
+      let date = !!dateMatches && dateMatches.length > 0 ? dateMatches[0].replaceAll('-','/') : '';
+      let descriptionMatches = page.url.match(/[A-Za-z]+-[A-Za-z]+/);
+      let description = !!descriptionMatches && descriptionMatches.length > 0 ? descriptionMatches[descriptionMatches.length - 1] + '/' : '';
+      return `/${date}/${description}`;
+    }
+  });
+
```

Comment:
I think these might be better done using folder-level defaults values in the directory data files in eleventy: https://www.11ty.dev/docs/data-template-dir/